llama_index/indices/struct_store/sql_query.py

```diff
@@ -203,6 +206,134 @@ async def _aquery(self, query_bundle: QueryBundle) -> Response:
         return response
 
 
+class SQLTableQueryEngine(BaseQueryEngine):
+    def __init__(
+        self,
+        index: SQLTableIndex,
+        **kwargs: Any,
+    ) -> None:
+        self._sql_database = index.sql_database
+        super().__init__(index.service_context.callback_manager)
+
+    def _query(self, query_bundle: QueryBundle) -> Response:
+        """Answer a query."""
+        # NOTE: override query method in order to fetch the right results.
+        # NOTE: since the query_str is a SQL query, it doesn't make sense
+        # to use ResponseBuilder anywhere.
+        response_str, extra_info = self._sql_database.run_sql(query_bundle.query_str)
+        response = Response(response=response_str, extra_info=extra_info)
+        return response
+
+    async def _aquery(self, query_bundle: QueryBundle) -> Response:
+        return self._query(query_bundle)
+
+
+class NLSQLTableQueryEngine(BaseQueryEngine):
+    def __init__(
+        self,
+        index: SQLTableIndex,
```

Comment:
i wonder if we should create two NL query engines. 
The first doesn't take in an index at all, just the sql_database. and _get_table_context just dumps the table schema from the sql database class.

The second actually does retrieval from the table index.

Reply:
Hmm the retrieval is actually happening in SQLTableRetriever though? The NLQueryEngine mainly orchestrates the NL to SQL prediction.

Reply:
yeah but if you pass in sql_database directly then you can just do `sql_database.table_info` and dump that into the prompt. easier ergonomics for user if they're not worreid about retrieval